packages/cactus-test-plugin-htlc-eth-besu/src/test/typescript/integration/plugin-htlc-eth-besu/refund-endpoint.test.ts

```diff
@@ -0,0 +1,460 @@
+      secret: besuKeyPair.privateKey,
+      type: Web3SigningCredentialType.PRIVATEKEYHEX,
+    },
+    bytecode: DemoHelpers.bytecode,
+    gas: dataTest.estimated_gas,
+  });
+  t.comment(deployDemoHelper.transactionReceipt.contractAddress!);
+  t.ok(deployDemoHelper, "deployContract DemoHelpers is OK");
+
+  let timestamp: number;
+  const { callOutput } = await connector.invokeContract({
+    contractAbi: DemoHelpers.abi,
+    contractAddress: deployDemoHelper.transactionReceipt
+      .contractAddress as string,
+    invocationType: EthContractInvocationType.CALL,
+    methodName: "getTimestamp",
+    params: [],
+    web3SigningCredential: {
+      ethAccount: firstHighNetWorthAccount,
+      type: Web3SigningCredentialType.PRIVATEKEYHEX,
+      secret: besuKeyPair.privateKey,
+    },
+    gas: dataTest.estimated_gas,
+  });
+  t.ok(callOutput, "getTimestamp is truthy OK");
+  timestamp = callOutput as number;
+  timestamp = +timestamp + +10;
+
+  const balance1 = await web3.eth.getBalance(firstHighNetWorthAccount);
+  t.comment(balance1);
+
+  //newContract
+  const bodyObj: NewContractObj = {
+    contractAddress: hashTimeLockAddress,
+    inputAmount: 10,
+    outputAmount: 1,
+    expiration: timestamp,
+    hashLock: dataTest.hashLock,
+    receiver: dataTest.receiver,
+    outputNetwork: "BTC",
+    outputAddress: "1AcVYm7M3kkJQH28FXAvyBFQzFRL6xPKu8",
+    keychainId: keychainId,
+    connectorId: connectorId,
+    accountRef: accountRef,
+    privateKeyRef: privateKeyRef,
+    credentialType: Web3SigningCredentialType.PRIVATEKEYHEX,
+    gas: dataTest.estimated_gas,
+  };
+  const resp = await api.newContract(bodyObj);
+  t.ok(resp, "new contract");
+
+  const resp2 = await connector.invokeContract({
+    contractAbi: DemoHelpers.abi,
+    contractAddress: deployDemoHelper.transactionReceipt
+      .contractAddress as string,
+    invocationType: EthContractInvocationType.CALL,
+    methodName: "getTxId",
+    params: [
+      firstHighNetWorthAccount,
+      dataTest.receiver,
+      10,
+      dataTest.hashLock,
+      timestamp,
+    ],
+    web3SigningCredential: {
+      ethAccount: firstHighNetWorthAccount,
+      type: Web3SigningCredentialType.PRIVATEKEYHEX,
+      secret: besuKeyPair.privateKey,
+    },
+  });
+  t.ok(resp2.callOutput, "result invoke is truthy OK");
+  const id = resp2.callOutput as string;
+  t.comment(id);
+  const balance = await web3.eth.getBalance(firstHighNetWorthAccount);
+  t.comment(balance);
+  t.equal(
+    parseInt(balance),
+    parseInt(balance1) - 10,
+    "The balance of accout is OK",
+  );
+
+  // Test for invalid response test case
+  try {
+    const refundRequest: RefundReq = {
+      contractAddress: hashTimeLockAddress,
+      credentialType: Web3SigningCredentialType.PRIVATEKEYHEX,
+      keychainId,
+      connectorId,
+      accountRef,
+      privateKeyRef,
+    };
+    const res = await api.refund(id, refundRequest);
+    t.equal(res.status, 500, "response status is 500");
```

Comment:
@AzaharaC Code 500 should be returned for unknown errors. If the server validated the request and found that it had problems (such as something not being found that the request referred to as something in existence or something being in an invalid format) then a 400 or some other 4xx code could better indicate to end users that they need to fix something in their request.
In general 5xx codes mean that the backend is "at fault" meaning that the developer who sent the request has no action need, but in this case they do (which can be expressed in general with some 4xx codes that in general means that the backend thinks that the request sender is at fault)

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

Reply:
Done